### PR TITLE
#239 UWP Log View Redesign

### LINKED ISSUE #239: 💙 - UWP Log view redesign
> 🟥🟩 **WINDOWS 10** - Version 1903+ is required for the UWP UI.
> 🟦🟨 [How to setup UWP](https://docs.microsoft.com/en-us/windows/uwp/get-started/get-set-up)

## Summary

Our current log view in the Settings section of our UWP app uses the default list view component.

image

However, that provides features that we don't really need, such as item selection, hover animation, etc. Not to mention it takes too much space.

Ideally, we'd get something simpler, perhaps even a plain old readonly rich textbox.

The goal is to make it simple and usable as a log. So as long as the user can select all contents of the log and copy them, it's ideal. 🙂 

## How to implement

This change is entirely in the Settings section of our UWP application.

## Interested? 😊 

If you'd like to help us by solving this issue, just say so in the comments. ❤️ 
And don't shy away from asking any kind of question you might have.

Should you have any questions at all, feel free to contact either me (@petrspelos) or @DraxCodes on our Discord server:

[Discord](https://discord.gg/cGhEZuk)

_And of course, thank you so much for contributing! 🎉 It's because of amazing people like yourself that Miunie exists._

### PR DESCRIPTION
**Pull request by 1n5an1ty**
## Related Issue

Fixes #239 

## Changes
Update EnterKeyHelpers to work with PasswordBox
Update Page style for Settings to display logs as selectable textblock
Add 'Enter' key trigger command for applying token

## Expected Feedback
General 5an1ty checks

### CODE REVIEW

File: src/Miunie.WindowsApp/ViewModels/SettingsPageViewModel.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public SettingsPageViewModel(MiunieBot miunie, ILogReader logReader, TokenManage
 
         public string BotAvatar => _miunie.MiunieDiscord.GetBotAvatarUrl() ?? DefaultAvatarUrl;
 
-        public IEnumerable<object> Logs => _logReader.RetrieveLogs(10).Select(m => new { Message = m });
+        public string Logs => string.Join(Environment.NewLine, _logReader.RetrieveLogs(10).Select(m => $"{m}"));
```
Comment (petrspelos) [inline src/Miunie.WindowsApp/ViewModels/SettingsPageViewModel.cs]: I thought `RetrieveLogs` returns an `IEnumerable<string>`. In that case why do we have that `Select` statement there when the result of that is also an `IEnumerable<string>`? 🤔 

Is there something I'm missing?

File: src/Miunie.WindowsApp/Views/SettingsPage.xaml [hunk 2]
```diff
@@ -4,17 +4,19 @@
...
 
     <Grid Padding="32">
         <Grid.RowDefinitions>
+            <RowDefinition Height="auto"/>
             <RowDefinition Height="auto"/>
             <RowDefinition Height="*"/>
         </Grid.RowDefinitions>
         <Grid.ColumnDefinitions>
-            <ColumnDefinition Width="auto"/>
+            <ColumnDefinition Width="auto" />
```
Comment (petrspelos) [inline src/Miunie.WindowsApp/Views/SettingsPage.xaml]:
```suggestion
            <ColumnDefinition Width="auto"/>
```

File: src/Miunie.WindowsApp/ViewModels/SettingsPageViewModel.cs (hunk 1 above)
Comment (1n5an1ty) [reply src/Miunie.WindowsApp/ViewModels/SettingsPageViewModel.cs]: Nope, you are not missing anything there. It was leftover code from a refactor where I was hoping to get a DateTime stamp but only IEnum of string. Removed!

**Comment by petrspelos** [comment]
image

Looks really good on my Windows 10 (1909), thanks @1n5an1ty! Works as expected. ❤️

**Pull request merged**